src/router/queue.router.ts

```diff
@@ -38,8 +41,29 @@ export class QueueRouter extends AbstractRouter {
         });
     }
 
-    private getQueues(request: Request) {
-        return this.queueManager.getAll();
+    private async getQueues(request: Request) {
+        const currentPage: number = +request.query.currentPage;
+        const itemsPerPage: number = +request.query.itemsPerPage;
+        const sortBy: string = request.query.sortBy as string;
+        const sortOrder: number = +request.query.sortOrder;
+        
+        if (currentPage && itemsPerPage) {
+            let data: responseData = await this.queueManager.getManyByOptions(currentPage, itemsPerPage, sortBy, sortOrder);
+            const jobs: Array<Queue> = this.queueService.getJobs();
+
+            data.list.forEach(item => {
```

Comment:
the full logic can be done in the service and the result returned directly, to keep everything light:

return this.queueService.applyProgressToJobs(data)